### ISSUE TITLE
[INFRA-3] ExcelDataLoader + Data Population

### ISSUE THREAD

**Issue by chuadongen** [JacobSoh/cs102-g4-t6#27]
## Description
Implement Excel data loading using Apache POI. Populate with all Splendor card data.

## Tasks

- [x] Implement `ExcelDataLoader.java`:
  - Private constructor (utility class)
  - `static loadLevel1Cards(filePath: String): List<Card>`
  - `static loadLevel2Cards(filePath: String): List<Card>`
  - `static loadLevel3Cards(filePath: String): List<Card>`
  - `static loadNobles(filePath: String): List<Noble>`
  - Private helper method: `loadCardsFromSheet(filePath, sheetName, level)`
- [ ] Use Apache POI (XSSFWorkbook, Sheet, Row)
- [x] Handle IOException with RuntimeException and clear message
- [x] All methods have JavaDoc comments
- [x] Test with sample data (5 cards)
- [x] Verify splendor_data.xlsx exists (should already be populated)
- [ ] Verify all 4 sheets exist:
  - Level1Cards (40 cards)
  - Level2Cards (30 cards)
  - Level3Cards (20 cards)
  - Nobles (10 nobles)
- [ ] Spot-check 10 random cards against official Splendor rules
- [ ] Fix any data errors found

### Both: Integration (Tuesday)
- [ ] Test loader with full data file
- [ ] Verify all 90 cards + 10 nobles load correctly
- [ ] Print sample output to verify
- [ ] Create feature branch: `feature/excel-loader`

## Example Code Structure
```java
public class ExcelDataLoader {
    private ExcelDataLoader() {}
    
    public static List<Card> loadLevel1Cards(String filePath) {
        return loadCardsFromSheet(filePath, "Level1Cards", 1);
    }
    
    private static List<Card> loadCardsFromSheet(String filePath, String sheetName, int level) {
        List<Card> cards = new ArrayList<>();
        
        try (FileInputStream fis = new FileInputStream(filePath);
             Workbook workbook = new XSSFWorkbook(fis)) {
            
            Sheet sheet = workbook.getSheet(sheetName);
            
            for (int i = 1; i <= sheet.getLastRowNum(); i++) {  // Skip header
                Row row = sheet.getRow(i);
                
                // Read: Level | Points | Bonus | CostWhite | CostBlue | ...
                int points = (int) row.getCell(1).getNumericCellValue();
                GemColor bonus = GemColor.valueOf(row.getCell(2).getStringCellValue());
                
                Map<GemColor, Integer> costMap = new EnumMap<>(GemColor.class);
                costMap.put(GemColor.WHITE, (int) row.getCell(3).getNumericCellValue());
                // ... read other colors
                
                Cost cost = new Cost(costMap);
                cards.add(new Card(level, points, bonus, cost));
            }
        } catch (IOException e) {
            throw new RuntimeException("Failed to load cards: " + e.getMessage(), e);
        }
        
        return cards;
    }
}
```

## Acceptance Criteria
- [x] ExcelDataLoader loads all cards and nobles correctly
- [ ] Excel file has all data (verified)
- [x] All methods have JavaDoc comments
- [ ] Test output shows correct counts (40, 30, 20, 10)
- [ ] Sample cards print correctly
- [ ] PR and ping me dawg

## Dependencies
- Blocked by legit everything

## Labels
`infrastructure` `data` `sprint-1` `day-5-6`
